backend/src/schema/resolvers/searches/queryString.spec.js

```diff
@@ -0,0 +1,43 @@
+import { queryString, escapeSpecialCharacters, normalizeWhitespace } from './queryString'
+
+describe('queryString', () => {
+  describe('special characters', () => {
+    it('does escaping correctly', () => {
+      expect(escapeSpecialCharacters('+ - && || ! ( ) { } [ ] ^ " ~ * ? : \\ / ')).toEqual(
+        '\\+ \\- \\&\\& \\|\\| \\! \\( \\) \\{ \\} \\[ \\] \\^ \\" \\~ \\* \\? \\: \\\\ \\/ ',
+      )
+    })
+  })
+
+  describe('whitespace', () => {
+    it('normalizes correctly', () => {
+      expect(normalizeWhitespace(' a \t \n b \n   ')).toEqual('a b')
+    })
+  })
+
+  describe('exact match', () => {
+    it('boosts score by factor 8', () => {
+      expect(queryString('a couple of words')).toContain('"a couple of words"^8')
+    })
+  })
+
+  describe('match all words exactly', () => {
+    it('boosts score by factor 4', () => {
+      expect(queryString('a couple of words')).toContain(
+        '("a" AND "couple" AND "of" AND "words")^4',
+      )
+    })
+  })
+
+  describe('match at least one word exactly', () => {
+    it('boosts score by factor 2', () => {
+      expect(queryString('a couple of words')).toContain('"a"^2 "couple"^2 "of"^2 "words"^2')
+    })
+  })
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-03-17T09:48:43Z" title="Tuesday, March 17th 2020, 10:48:43 am +01:00">Mar 17, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,43 @@
+import { queryString, escapeSpecialCharacters, normalizeWhitespace } from './queryString'
+
+describe('queryString', () => {
+  describe('special characters', () => {
+    it('does escaping correctly', () => {
+      expect(escapeSpecialCharacters('+ - && || ! ( ) { } [ ] ^ " ~ * ? : \\ / ')).toEqual(
+        '\\+ \\- \\&\\& \\|\\| \\! \\( \\) \\{ \\} \\[ \\] \\^ \\" \\~ \\* \\? \\: \\\\ \\/ ',
+      )
+    })
+  })
+
+  describe('whitespace', () => {
+    it('is normalized correctly', () => {
+      expect(normalizeWhitespace(' a \t \n b \n   ')).toEqual('a b')
+    })
+  })
+
+  describe('exact match', () => {
+    it('boosts score by factor 8', () => {
+      expect(queryString('a couple of words')).toContain('"a couple of words"^8')
+    })
+  })
+
+  describe('match all words exactly', () => {
+    it('boosts score by factor 4', () => {
+      expect(queryString('a couple of words')).toContain(
+        '("a" AND "couple" AND "of" AND "words")^4',
+      )
+    })
+  })
+
+  describe('match at least one word exactly', () => {
+    it('boosts score by factor 2', () => {
+      expect(queryString('a couple of words')).toContain('"a"^2 "couple"^2 "of"^2 "words"^2')
+    })
+  })
```


I guess these tests give a hint of how the boosting works... I would infer that 8 is the maximum and you can evenly distribute that amongst words?

Reply:
> [<img alt="Mogge" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Mogge) **Authored by [Mogge](https://github.com/Mogge)**
_<time datetime="2020-03-17T11:17:51Z" title="Tuesday, March 17th 2020, 12:17:51 pm +01:00">Mar 17, 2020</time>_

---


As far as I know, there is no maximum.